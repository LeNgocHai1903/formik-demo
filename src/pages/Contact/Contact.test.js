import React from 'react'
import renderer from 'react-test-renderer'
import {render, fireEvent} from '@testing-library/react'
import  Contact from './Contact'
import { act } from 'react-dom/test-utils';
import { unmountComponentAtNode } from 'react-dom';

let container = null;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
  });
  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

describe('Contact Component', () => {
    it('should render component correctly', () => {
        const component = renderer.create(<Contact/>).toJSON();
        expect(component).toMatchSnapshot();
    })

    it('changes value when clicked', async () => {
        const { getByTestId} = render(<Contact />)
        const name = getByTestId("yourName")
        act(() => {
            fireEvent.blur(name)
        });
        await (() => {
            expect(getByTestId("errorMessage")).toHaveTextContent('Required!');
        })

    })
    it('should submit correct information', async () => {
        const { getByTestId} = render(<Contact />)
        const name = getByTestId("yourName")
        const selectVal = getByTestId("select")
        const message = getByTestId("message")
        const formContact = getByTestId("form-contact")

        const fakeInfo = {
            fakeName: 'Le Hai',
            fakeSelect: 'Rider',
            fakeMessage: 'Unit Testing'
        }
        act(() => {
            fireEvent.change(name, {target: {value: fakeInfo.fakeName}})
            fireEvent.change(selectVal, {target: {value: fakeInfo.fakeSelect}})
            fireEvent.change(message, { target: {value: fakeInfo.fakeMessage}})
            fireEvent.submit(formContact)
        });
        await (() => {
            expect(name).toHaveTextContent(fakeInfo.fakeName);
            expect(selectVal).toHaveTextContent(fakeInfo.fakeSelect)
            expect(message).toHaveTextContent(fakeInfo.fakeMessage)
        })
    })

})
