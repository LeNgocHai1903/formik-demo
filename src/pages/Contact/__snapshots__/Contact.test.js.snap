// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component should render component correctly 1`] = `
<div
  className="SignUp"
>
  <h1>
    <i
      aria-hidden="true"
      className="fa fa-envelope"
    />
     Contact Us
  </h1>
  <form
    data-testid="form-contact"
    onSubmit={[Function]}
  >
    <div
      className="FormStyle"
    >
      <div
        className="LeftInfo"
      >
        <div
          className="input-group mb-2"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
              id="inputGroup-sizing-default"
            >
              Your Name:
            </span>
          </div>
          <input
            aria-describedby="inputGroup-sizing-default"
            aria-label="Default"
            className="form-control"
            data-testid="yourName"
            id="contactName"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
          />
        </div>
        <div
          className="input-group mb-2"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
              id="inputGroup-sizing-default"
            >
              Select:
            </span>
          </div>
          <select
            aria-describedby="inputGroup-sizing-default"
            aria-label="Default"
            className="form-control"
            data-testid="select"
            name="select"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
          >
            <option
              value=""
            >
              Choose...
            </option>
            <option
              value="Example - 1"
            >
              Example - 1
            </option>
            <option
              value="Example - 2"
            >
              Example - 2
            </option>
            <option
              value="Example - 3"
            >
              Example - 3
            </option>
          </select>
        </div>
      </div>
      <div
        className="RightInfo"
      >
        <div
          className="input-group mb-2"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
              id="inputGroup-sizing-default"
            >
              Message:
            </span>
          </div>
          <textarea
            aria-describedby="inputGroup-sizing-default"
            aria-label="Default"
            className="form-control"
            data-testid="message"
            name="message"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
          />
        </div>
      </div>
    </div>
    <button
      className="submitBtn"
      disabled={false}
      type="submit"
    >
      <i
        aria-hidden="true"
        className="fa fa-2x fa-arrow-right"
      />
    </button>
  </form>
</div>
`;
